CMakeLists.txt

```diff
@@ -517,10 +517,8 @@ if(MSVC)
   include_directories(SYSTEM src/platform/msc)
 else()
   include(TestCXXAcceptsFlag)
-  if (NOT ARM6)
-    if(NOT DEPENDS OR DEPENDS AND NOT ARM)
-      set(ARCH native CACHE STRING "CPU to build for: -march value or 'default' to not pass -march at all")
-    endif()
+  if (NOT ARCH)
+    set(ARCH native CACHE STRING "CPU to build for: -march value or 'default' to not pass -march at all")
```

Comment:
Why did you take out the DEPENDS check ?

Reply:
This check is the main reason we have 0.13 builds broken on armv7.
We don't need it. I think this kind of thing should be done on the Travis or Depends side.

This check is only needed for Travis+Depends darwin build, but take a look what it actually does:
1. Depends is invoked before the Monero's CMake
2. Depends sets ARCH='x86_64'
3. Monero's CMake invoked
4. Due to this check, Monero's CMake reverts  ARCH to ARCH='native'
5. We compile so-called darwin x86_64 build with arch 'native'. This doesn't make any sense. Is not actually a cross-compilation.

Reply:
@TheCharlatan ping

Reply:
Thanks for taking this apart. I agree with the assessment, can you instead add a flag in contrib/depends/toolchain.cmake.in to activate when building darwin11 and check that in the line you are removing, so it does not break current builds?

Reply:
@moneromooo-monero @TheCharlatan updated

Reply:
Depends will not compile at all, if the architecture flag is hard set to native, because it cross compiles everything. This does not just count for the Travis runs, but for any use that depends has, including deterministic builds. I think it might be better to remove the set(ARCH ...) and instead check for armv7 and armv8 as well and then set the arch flag manually, instead of imposing it over the entire build.